cli/azd/pkg/tools/azcli/deployments.go

```diff
@@ -75,3 +78,127 @@ func (cli *azCli) createDeploymentsClient(
+		return AzCliDeploymentResult{}, fmt.Errorf("reading parameters file: %w", err)
+	}
+
+	createFromTemplateOperation, err := deploymentClient.BeginCreateOrUpdateAtSubscriptionScope(
+		ctx, deploymentName,
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   templateJsonAsMap,
+				Parameters: parametersFileJsonAsMap["parameters"],
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+			Location: to.Ptr(location),
+		}, nil)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("starting deployment to subscription: %w", err)
+	}
+
+	// wait for deployment creation
+	deployResult, err := createFromTemplateOperation.PollUntilDone(ctx, nil)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("deploying to subscription: %w", err)
+	}
+
+	return AzCliDeploymentResult{
+		Properties: AzCliDeploymentResultProperties{
+			Outputs: CreateDeploymentOutput(deployResult.Properties.Outputs),
+		},
+	}, nil
+}
+
+func (cli *azCli) DeployToResourceGroup(
+	ctx context.Context, subscriptionId, resourceGroup, deploymentName, compiledBicep, parametersFile string) (
+	AzCliDeploymentResult, error) {
+	deploymentClient, err := cli.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	templateJsonAsMap, err := readFromString([]byte(compiledBicep))
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("reading template file: %w", err)
+	}
+	parametersFileJsonAsMap, err := readJson(parametersFile)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("reading parameters file: %w", err)
+	}
+
+	createFromTemplateOperation, err := deploymentClient.BeginCreateOrUpdate(
+		ctx, resourceGroup, deploymentName,
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   templateJsonAsMap,
+				Parameters: parametersFileJsonAsMap["parameters"],
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+		}, nil)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("starting deployment to resource group: %w", err)
+	}
+
+	// wait for deployment creation
+	deployResult, err := createFromTemplateOperation.PollUntilDone(ctx, nil)
+	if err != nil {
+		return AzCliDeploymentResult{}, fmt.Errorf("deploying to resource group: %w", err)
+	}
+
+	return AzCliDeploymentResult{
+		Properties: AzCliDeploymentResultProperties{
+			Outputs: CreateDeploymentOutput(deployResult.Properties.Outputs),
+		},
+	}, nil
+}
+
+func readJson(path string) (map[string]interface{}, error) {
+	templateFile, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return readFromString(templateFile)
+}
+
+func readFromString(jsonBytes []byte) (map[string]interface{}, error) {
+	template := make(map[string]interface{})
+	if err := json.Unmarshal(jsonBytes, &template); err != nil {
+		return nil, err
+	}
+
+	return template, nil
+}
```

Comment:
Consider adding these into our `convert` package so others can leverage converting from strings to maps

Reply:
I'd rather wait until someone else need it, so we can move it